aiohttp/client_reqrep.py

```diff
@@ -422,22 +421,20 @@ def update_cookies(self, cookies: Optional[LooseCookies]) -> None:
 
         self.headers[hdrs.COOKIE] = c.output(header="", sep=";").strip()
 
-    def update_content_encoding(self, data: Any) -> None:
+    def update_content_encoding(self, data: Any, compress: Union[bool, str]) -> None:
         """Set request content encoding."""
+        self.compress = None
         if not data:
-            # Don't compress an empty body.
-            self.compress = None
             return
 
         enc = self.headers.get(hdrs.CONTENT_ENCODING, "").lower()
         if enc:
```

Comment:
```suggestion
        if enc := self.headers.get(hdrs.CONTENT_ENCODING):
```

I think there is no reason to call `.lower()` here if we only need to see if its set

Reply:
This is an existing issue though.  I'll do a new PR so we can backport

Reply:
https://github.com/aio-libs/aiohttp/pull/9234